apps/05Mastermind.js

```diff
@@ -26,12 +27,36 @@ function getRandomInt(min, max) {
   return Math.floor(Math.random() * (max - min)) + min;
 }
 
-function generateHint() {
-    // your code here
+function generateHint(solution, guess) {
+    var solutionArray = solution.split('');
+    var guessArray = guess.split('');
+    var correctLetterLocations = 0;
+    for (var i=0; i < 4; i++) {
+        if(solutionArray[i] === guessArray[i]) {
+            correctLetterLocations++;
+            solutionArray[i] = null;
+        }
+    }
+    var correctLetters = 0;
+    for (var i=0; i < 4; i++) {
+        if (solutionArray[i] === guess[0] || solutionArray[i] === guess[1] || solutionArray[i] === guess[2] || solutionArray[i] === guess[3]) {
```

Comment:
While this functions, it is not very clean/efficient. Remember you can use the indexOf on an array to see if something is in that array. In this case, we could use indexOf on the guessArray to see if that particular element of the solutionArray exists in the guess array.:

`
if(guessArray.indexOf(solutionArray[i]) != -1){
}
`